esmvalcore/cmor/_fixes/cordex/cordex_fixes.py

```diff
@@ -166,6 +182,177 @@ def _fix_geographical_coords(self, cube, domain):
+        # Transform std domain points in lambert to compare.
+        lambert_bounds = self._transform_points(
+            GEOG_SYSTEM, cube.coord_system().as_cartopy_crs(),
+            np.array(domain_bounds['lons']),
+            np.array(domain_bounds['lats']))
+
+        proj_x = cube.coord(var_name='x')
+        proj_y = cube.coord(var_name='y')
+        proj_x.convert_units('m')
+        proj_y.convert_units('m')
+
+        # Compare the 4 corners of cube domain vs. the standard domain ones.
+        cube_corners = np.take(lambert_bounds, indices=[0, 2, 6, 8], axis=0)
+        domain_corners = np.array([
+            [proj_x.points.min(), proj_y.points.max()],
+            [proj_x.points.max(), proj_y.points.max()],
+            [proj_x.points.min(), proj_y.points.min()],
+            [proj_x.points.max(), proj_y.points.min()]])
+        # Check: euclidean distance between cube domain and std domain < 1e6.
+        check = np.mean(
+            np.linalg.norm(cube_corners - domain_corners, axis=1)) < 1e6
+
+        # Check passes -> proj coords are good -> fix and plug metadata.
+        if check:
+            self._replace_coord(
+                cube, proj_x, self._make_projection_coord(
+                    proj_x.points,
+                    CMOR_TABLES["CORDEX"].coords["x"],
+                    cube.coord_system()
+                ))
+            self._replace_coord(
+                cube, proj_y, self._make_projection_coord(
+                    proj_y.points,
+                    CMOR_TABLES["CORDEX"].coords["y"],
+                    cube.coord_system()
+                ))
+        return check
+
+    def _check_lambert_conformal_geog_coords(self, cube, domain_bounds):
+        """Check lambert conformal geographical coordinates."""
+        dom_lons = np.array(domain_bounds['lons'])
+        dom_lats = np.array(domain_bounds['lats'])
+        # Check without time coordinate if present.
+        cube2d = cube[0] if 'time' in self.vardef.dimensions else cube
+        # Crops cube on the standard domain.
+        cube_x_std = extract_region(cube2d,
+                                    start_longitude=dom_lons.min(),
+                                    end_longitude=dom_lons.max(),
+                                    start_latitude=dom_lats.min(),
+                                    end_latitude=dom_lats.max())
+        # Check: <15% diff in size between original cube and cropped.
+        check = (np.sum(cube2d.shape) / np.sum(cube_x_std.shape)) < 1.15
+        # Check passes -> geog coords are good -> fix and plug metadata.
+        if check:
+            for var_name in ["longitude", "latitude"]:
+                old_coord = cube.coord(var_name)
+                self._replace_coord(
+                    cube, old_coord, self._make_geographical_coord(
+                        old_coord.points,
+                        CMOR_TABLES["CORDEX"].coords[var_name],
+                        old_coord.bounds
```

Comment:
What if the old coordinate does not have bounds?

Reply:
The checks will fix the metadata (and guess the bounds for the projection coords) that are used for the fix:
Notice that the function `_make_projection_coord` that I use to fix the metadata contains the instruction `coord.guess_bounds()` at the end.

Reply:
This is using `_make_geographical_coord`, because it's supposed to fix latitude and longitude, not the projection coordinates right?

Reply:
In order to check if the idea exposed above is reasonable I have been performing the following test:
- Idea:
use the regrid function from `esmvalcore.preprocessor` to test if the guessed bounds give visually correct results.

  - Testset 1: {model-> Aladin63, exp->rcp85, driver -> NCC-NorESM1-M, institute->CNRM, domain->EUR-11} 
  _Note_:  Coastlines don't show up because the domain is shifted, that is handled by other parts of the fix.
    - Initially has bounds -> I remove the bounds for `latitude` and `longitude`, make them again as explained above then plot the following scenarios: 
      - Original cube
![image](https://user-images.githubusercontent.com/19970598/224004174-711d771e-cbbb-4198-8e74-847c6319fe54.png)
      - Original cube regrided against itself
![image](https://user-images.githubusercontent.com/19970598/224004205-afbc7001-7b42-4d22-ae59-6db71676e22e.png)
      - Original cube regrided against guessed bounds cube
![image](https://user-images.githubusercontent.com/19970598/224004366-b0a76e40-0df8-4588-8b0b-742a47659418.png)
      - Guessed bounds cube regrided against original cube
![image](https://user-images.githubusercontent.com/19970598/224004870-acd511a5-b9aa-46f2-b9fc-3a44fe64f284.png)

  - Testset 2: {model-> Regcm4-6, exp->historical, driver -> MPI-M-MPI-ESM-LR, institute->ICTP,  domain->EUR-11}
  _Note_: the choice of 'pr' variable here might not be the best as there is not much happening visually, but in case of a faulty regridding I understood that some glitches were supposed to show.
    - Initially no bounds -> I make the bounds for geographical coords then plot the following:
      - Original cube
![image](https://user-images.githubusercontent.com/19970598/224005822-e832efd9-70d2-4ec5-b139-d8a0b47dc4ae.png)
      - Guessed bounds cube regrided against itself
![image](https://user-images.githubusercontent.com/19970598/224006230-ea77b4c6-60dd-44bc-821e-229928135815.png)
      - Guessed bounds cube regrided against Original cube of Testset 1
        - The regriding operation fails (for reasons that look unrelated), I could investigate deeper if you think this is an interesting experiment.
        - Also fails when trying the reverse operation 


- Conclusions:
Checking the visuals and also the shapes of the output cubes, everything seems preserved. I think the fix idea is good and will move forward to add it in the next version.
I am however not a climate scientist if someone could confirm me that everything looks good that would be very helpful.

Do not hesitate to ask me for more experiments, any kind of operation making use of the bounds of the geographical coordinates could be interesting to try out I think.

Reply:
call to `_check_lonlat_bounds` function added at the end of `fix_metadata`